src/main/java/teammates/ui/automated/PrepareEmailWorkerAction.java

```diff
@@ -0,0 +1,163 @@
+package teammates.ui.automated;
+
+import java.util.List;
+
+import teammates.common.datatransfer.CommentSendingState;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.exception.TeammatesException;
+import teammates.common.util.Assumption;
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HttpRequestHelper;
+import teammates.common.util.Const.ParamsNames;
+import teammates.logic.core.CommentsLogic;
+import teammates.logic.core.EmailGenerator;
+import teammates.logic.core.EmailSender;
+import teammates.logic.core.FeedbackResponseCommentsLogic;
+import teammates.logic.core.FeedbackSessionsLogic;
+
+/**
+ * Task queue worker action: prepares large number of emails to be sent via task queue.
+ */
+public class PrepareEmailWorkerAction extends AutomatedAction {
```

Comment:
This one has a large change when migrated from `EmailAction`, both open for discussion:
- Most of the logging activities are removed, because I cannot really find how they value-add over the existing `log.severe`s and `log.info`s.
- The `*MailAction` present before follows the template pattern (which I have been championing for the past few PRs), but they created more duplications than they have done abstractions (don't ask me how/why...). I am quite conflicted over choosing the design for this one, but I tend more towards keeping it as `switch-case`s for now as the number of cases is not big enough.